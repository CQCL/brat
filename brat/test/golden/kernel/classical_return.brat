-- This should fail because it's invalid to have an `Int` in a kernel context.
-- As it stands, this doesn't even parse because once the parser thinks it's
-- looking at a kernel, it only looks for valid kernel types like `Qubit` and
-- `Bool`. We could probably catch this and make the error message nicer.
f :: { Qubit -o Qubit, Int }
f = { q -> q, 1 }
