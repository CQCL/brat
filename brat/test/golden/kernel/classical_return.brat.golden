Error in test/golden/kernel/classical_return.brat@FC {start = Pos {line = 4, col = 4}, end = Pos {line = 4, col = 6}}:
  Parse error test/golden/kernel/classical_return.brat:5:6:
  |
5 | f :: { Qubit -o Qubit, Int }
  |      ^^^^^^^^^^^^^^^^^^^^^^^
unexpected { Qubit -o Qubit, Int }
expecting (...), ->, -o, Bit, Bool, Comment, Float, Horizontal space, Int, List, Money, Nat, Option, Pair, Qubit, String, Type, Unit, Vec, \n, identifier, name, whitespace, or {...}


